python_modules/dagster/dagster/_core/pipes/utils.py

```diff
@@ -323,40 +337,89 @@ def _messages_thread(
         self,
         handler: "PipesMessageHandler",
         params: PipesParams,
-        is_task_complete: Event,
+        is_session_closed: Event,
     ) -> None:
         start_or_last_download = datetime.datetime.now()
         while True:
             now = datetime.datetime.now()
-            if (now - start_or_last_download).seconds > self.interval or is_task_complete.is_set():
+            if (now - start_or_last_download).seconds > self.interval or is_session_closed.is_set():
                 start_or_last_download = now
                 chunk = self.download_messages_chunk(self.counter, params)
                 if chunk:
                     for line in chunk.split("\n"):
                         message = json.loads(line)
                         handler.handle_message(message)
                     self.counter += 1
-                elif is_task_complete.is_set():
+                elif is_session_closed.is_set():
+                    break
+            time.sleep(DEFAULT_SLEEP_INTERVAL)
+
+    def _logs_thread(
+        self,
+        params: PipesParams,
+        is_session_closed: Event,
+        are_all_messages_read: Event,
```

Comment:
maybe `reader_thread_closed`  ? `are_all_messages_read` is the expected state but "the message reader thread has been joined" is the more truthful expression (exception in reader thread could have led to successful join but not actually reading all messages)

Reply:
I changed it to just take the messages thread directly and call `is_alive()` on it.